### ISSUE TITLE
macOS & iOS support

### ISSUE THREAD

**Issue by alexmercerind** [alexmercerind/dart_vlc#242]
I cannot afford a mac or iPhone to support macOS & iOS on [`dart_vlc`](https://github.com/alexmercerind).

Current macOS & iOS implementation is written entirely by third-party contributors (uses same [`core`](https://github.com/alexmercerind/dart_vlc/tree/master/core) as Windows & Linux).
It needs updates when some internal API is changed or a new feature is added. Apart from this, we need some improvements to how video frames are transferred (prevent using CPU-based pixel buffers).
Since both Flutter for Desktop & [`dart_vlc`](https://github.com/alexmercerind) are new, a lot of time & effort is needed to make things stable & performant.

Thus, drop a `Hi!` here if you are interested in maintaining macOS & iOS support. I'll make sure to mention you in the project's page.
Skills required:
- **Hardware:** A macOS running device.
- **Minimum knowledge:** Objective-C, C++, C, Dart & Flutter.
- **Good to have knowledge of (not necessary but needed):** CMake, CocoaPods, dynamic/static linking, libvlc, Dart FFI, Flutter platform channels/texture APIs.
 

Thanks!

**Comment by ollydixon** [comment]
@alexmercerind if you cannot provide support, please remove that you support it on pub-dev.

<img width="560" alt="Screenshot 2022-06-10 at 12 52 37" src="https://user-images.githubusercontent.com/6940726/173050140-d5287b69-121d-418e-84ab-b9e26c7b07a6.png">

**Comment by ollydixon** [comment]
For anyone else who urgently needs a video player for MacOS. While we try to fix this one, you can use https://pub.dev/packages/video_player_macos

**Comment by GroovinChip** [comment]
iOS and macOS support has been dropped, per #303.

**Comment by felipecastrosales** [comment]
It would be awesome to have iOS support. :D

**Comment by mishkov** [comment]
> For anyone else who urgently needs a video player for MacOS. While we try to fix this one, you can use https://pub.dev/packages/video_player_macos

Yes, but this plugin does not support mkv files. At least via network

**Comment by ollydixon** [comment]
@mishkov indeed, we need someone to make a pull request.

**Comment by alexmercerind** [comment]
Just to inform you guys here, I'm no longer investing time in package:dart_vlc, but started off a new project package:media_kit.
Primarily because, package:dart_vlc or the package @ollydixon shared above copies every video frame back to CPU RAM from GPU for rendering, this means that your CPU cycles & battery life is wasted for no reason. [package:media_kit](https://github.com/alexmercerind/media_kit) renders video output directly from GPU (very good performance).

[package:media_kit](https://github.com/alexmercerind/media_kit) offers a very small bundle size as compared to package:dart_vlc, also it supports all the formats unlike @ollydixon's package.

> The primary goal of [package:media_kit](https://github.com/alexmercerind/media_kit) is to become a strong, stable, feature-proof & modular media playback library for Flutter. The idea is to support both audio & video playback.

See the issue: #345 for detailed about how video rendering works & how I've aimed to achieve better performance. The project [package:media_kit](https://github.com/alexmercerind/media_kit) is really-really ambitious & I need time.

Currently, Windows implementation is really performant, it can play 8K 60FPS videos very-very easily!
Expect Linux next week, macOS later this month.

Thanks!

**Comment by ollydixon** [comment]
@alexmercerind thanks for your hard work, this is very needed in the Flutter community. The fact that the flutter team hasn't fixed this is insane.

**Comment by ABDERRAHMANE-OUALI** [comment]
@alexmercerind thank you for your hard work, really need a video player on macOS
